libparsec/crates/client/tests/unit/certif/validate_child_manifest.rs

```diff
@@ -172,7 +174,7 @@ async fn corrupted(#[values("dummy", "parent_pointing_on_itself")] kind: &str, e
                 timestamp: now,
                 id: child_id,
                 parent: child_id,
-                version: 1,
```

Comment:
Remote manifest version is not supposed to be zero:
- A remote manifest is synchronized in the server, so it is accepted only if it version is > 0
- We don't enforce this in the type though given we use remote manifest within the local manifest where it can have `version == 0` if the local manifest is a placeholder.

Here the value is expected to be a valid one (given the error kind `parent_pointing_on_itself` doesn't mention anything about an invalid version), so I'd rather stick to `1`

Reply:
The problem is that this test wasn't actually testing that the internal error was actually corresponding to `parent_pointing_to_itself` (more specifically `DataError::Serialization`, which is a bit weird as well). So I improve the test and got the following error:
```
Diff < left / right > :
<InvalidManifest(
<    Corrupted {
<        realm: VlobID(
<            f0000000-0000-0000-0000-000000000002,
<        ),
<        vlob: VlobID(
<            a14d0099-ca88-447c-afd6-30383e1270ea,
<        ),
<        version: 0,
<        author: DeviceID(
<            "alice@dev1",
<        ),
<        timestamp: DateTime(
<            "2020-01-01T00:00:00Z",
<        ),
<        error: UnexpectedVersion {
<            expected: 0,
<            got: 1,
<        },
<    },
<)
>CertifValidateManifestError::InvalidManifest(boxed) if
>matches!(& * boxed, InvalidManifestError :: Corrupted { error, .. } if * *
>    error == DataError :: Serialization)
```
Since it said `UnexpectedVersion, expected 0 got 1` made the change without thinking about it. Isn't there something fishy here?